R/na_util.R

```diff
@@ -1,15 +1,28 @@
 #' Guess missing values by lm
 #' @export
-predict_na <- function(target, ...) {
-  # list(...) is a list of vectors
-  df <- as.data.frame(list(...))
-  df$target <- target
-  lm_model <- lm(data = df, target ~ ., na.action = na.omit)
-  ret <- target
-  ret[is.na(ret)] <- suppressWarnings({
-    # predict where target is NA
-    predict(lm_model, df[is.na(target),])
+impute_na <- function(target, ..., type = "lm_predict") {
+  ret <- switch(type, lm_predict = {
+    # list(...) is a list of vectors
+    df <- as.data.frame(list(...))
+    df$target <- target
+    lm_model <- lm(data = df, target ~ ., na.action = na.omit)
+    # predict return double, so values should be double
+    ret <- as.double(target)
```

Comment:
Shouldn't we convert lm return to target type rather than converting target to double?